### PR TITLE
Write Bash script to block deploys if client is not changed

### PR DESCRIPTION
**Pull request by tobias-mintlify**
Tested locally by running `bash shouldDeploy.sh` in the client folder. I updated the Vercel configuration to use the script but it may not take effect until the next deployment.

### CODE REVIEW

File: client/shouldDeploy.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [[ -z $(git diff --name-only HEAD~1 HEAD -- client) ]] ; then
+  echo "🛑 - Build cancelled. No changed to the client folder."
```
Comment (benjamin-shen) [inline client/shouldDeploy.sh:4]:
```suggestion
  echo "🛑 - Build cancelled. No changes to the client folder."
```

File: client/shouldDeploy.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  exit 0;
+fi
+
+if [[ "$VERCEL_ENV" == "production" ]] ; then
+  # Proceed with the build
```
Comment (benjamin-shen) [inline client/shouldDeploy.sh:9]: I think this comment might be redundant since the message is self-explanatory

**Review by benjamin-shen — COMMENTED** [review]
nice :)

**Review by hanywang2 — APPROVED** [review]
rubber stamped LGTM

**Pull request merged**